examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/UserLogProcessing.cs

```diff
@@ -0,0 +1,186 @@
+                
+            generalDf.Show();
+
+            // Only keep log entries that matched the reg ex
+            generalDf = generalDf.Filter(generalDf["GeneralReg(value, genfilter)"] == true);
+            generalDf.Show();
+            
+            // View the resulting schema
+            generalDf.PrintSchema();
+
+            // Step 2: Choose valid log entries that start with 10
+            spark.Udf().Register<string, string, bool>(
+                "IPReg", 
+                (log, type) => RegTest(log, type));
+                
+            generalDf.CreateOrReplaceTempView("IPLogs");
+            
+            // Apply UDF to get valid log entries start with 10
+            DataFrame ipDf = spark.Sql(
+                "SELECT iplogs.value, IPReg(iplogs.value, 'ipfilter') FROM IPLogs");
+                
+            ipDf.Show();
+
+            // Only keep log entries that matched both reg ex
+            ipDf = ipDf.Filter(ipDf["IPReg(value, ipfilter)"] == true); 
+            ipDf.Show();
+
+            // Step 3: Choose valid log entries that start 
+            // with 10 and deal with spam
+            spark.Udf().Register<string, string, bool>(
+                "SpamRegEx", 
+                (log, type) => RegTest(log, type));
+                
+            ipDf.CreateOrReplaceTempView("SpamLogs");
+            
+            // Apply UDF to get valid, start with 10, spam entries
+            DataFrame spamDF = spark.Sql(
+                "SELECT spamlogs.value, SpamRegEx(spamlogs.value, 'spamfilter') FROM SpamLogs");
+                
+            spamDF.Show();
+
+            // Only keep log entries that matched all 3 reg ex
+            DataFrame trueSpam = spamDF.Filter(spamDF["SpamRegEx(value, spamfilter)"] == true);
+            
+            // Formatting cleanup
+            // Use SQL to select just the entries, not boolean about reg ex
+            trueSpam.CreateOrReplaceTempView("TrueSpamLogs");
+            DataFrame trueSpamSql = spark.Sql(
+                "SELECT truespamlogs.value FROM truespamlogs");
+
+            // Explore the columns in the data we have filtered
+            // Let's try getting the number of GET requests
+            IEnumerable<Row> rows = trueSpamSql.Collect();
+            int numGetRequests = 0;
+            foreach(Row row in rows)
+            {
+                string rowstring = row.ToString();
+                numGetRequests += ParseLog(rowstring);
+            }
+
+            Console.WriteLine("Number of GET requests: " + numGetRequests);
+
+            spark.Stop();
+        }
+
+        public static bool RegTest(string logLine, string regexType)
+        {            
+            // Example Apache log line:   
+            // 64.242.88.10 - - [07/Mar/2004:16:47:12 -0800] "GET /robots.txt HTTP/1.1" 200 68
+            // 1:IP   2:client   3:user   4:date time   5:method   
+            // 6:req   7:proto   8:respcode   9:size
+            Regex useRx = new Regex(
```

Comment:
creating new Regex objects for every row is expensive.  Take a look at https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices#handle-object-instantiation-appropriately for best practices.

Reply:
Made the former regex objects now strings, and now using Regex.IsMatch() to compare 2 strings.